x/protocolpool/keeper/msg_server.go

```diff
@@ -103,11 +107,99 @@ func (k MsgServer) CommunityPoolSpend(ctx context.Context, msg *types.MsgCommuni
 }
 
 func (k MsgServer) CreateContinuousFund(ctx context.Context, msg *types.MsgCreateContinuousFund) (*types.MsgCreateContinuousFundResponse, error) {
+	if err := k.validateAuthority(msg.Authority); err != nil {
+		return nil, err
+	}
+
+	recipient, err := k.Keeper.authKeeper.AddressCodec().StringToBytes(msg.Recipient)
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate the message fields
+	err = k.validateContinuousFund(ctx, *msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create continuous fund proposal
+	cf := types.ContinuousFund{
+		Recipient:             msg.Recipient,
+		Percentage:            msg.Percentage,
+		MaxDistributedCapital: msg.MaxDistributedCapital,
+		Expiry:                msg.Expiry,
+	}
+
+	// Set continuous fund to the state
+	err = k.ContinuousFund.Set(ctx, recipient, cf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set recipient fund percentage & distribution
+	percentage := cf.Percentage.MulInt64(100)
```

Comment:
Isn't it easier to work with percentage from 0..1, instead of 0..100?

Reply:
yes, but we want to store this LegacyDec value in state. I don't know if LegacyDec is supported by collections, I used math.Int instead.